### PR TITLE
[plugin.video.vrt.nu] V1.5.1

### PR DESCRIPTION
**Pull request by pietje666**
### Description
- Refactored code
- Fixed categories that were not working anymore due to a scraping source update
- Fixed a subtitle issue
- Fixed a roaming issue

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.vrt.nu/resources/lib/kodiwrappers/kodiwrapper.py [hunk 1]
```diff
@@ -49,15 +49,18 @@ def play(self, video):
...
-            xbmc.Player().showSubtitles(True)
+            subtitles_visible = True
+            #separate subtitle url for hls-streams
             if video.subtitle_url is not None:
                 play_item.setSubtitles([video.subtitle_url])
-        else:
-            xbmc.Player().showSubtitles(False)
 
         xbmcplugin.setResolvedUrl(self._handle, True, listitem=play_item)
+        while not xbmc.Player().isPlaying():
```
Comment (enen92) [inline plugin.video.vrt.nu/resources/lib/kodiwrappers/kodiwrapper.py]: and not xbmc.Monitor().abortRequested()

File: plugin.video.vrt.nu/resources/lib/kodiwrappers/kodiwrapper.py [hunk 2]
```diff
@@ -49,15 +49,18 @@ def play(self, video):
...
+            xbmc.sleep(10)
```
Comment (enen92) [inline plugin.video.vrt.nu/resources/lib/kodiwrappers/kodiwrapper.py]: 10 ms is too low. Can you increase to 100?

File: plugin.video.vrt.nu/resources/lib/kodiwrappers/kodiwrapper.py [hunk 3]
```diff
@@ -71,12 +74,19 @@ def get_setting(self, setting_id ):
...
         self._addon.openSettings()
 
-    def check_inputstream_adaptive(self):
+    def has_inputstream_adaptive_installed(self): #note normally inputstream adaptive will always be installed, this only applies for people uninstalling inputstream adaptive while this addon is disabled
         return xbmc.getCondVisibility('System.HasAddon("{0}")'.format('inputstream.adaptive')) == 1
 
-    def check_widevine(self):
-        dirs, files = xbmcvfs.listdir(xbmc.translatePath('special://home/cdm'))
-        return any('widevine' in s for s in files)
+    def has_widevine_installed(self):
```
Comment (enen92) [inline plugin.video.vrt.nu/resources/lib/kodiwrappers/kodiwrapper.py]: You can use the inputstreamhelper script to help you setup widevine

**Comment by pietje666** [comment]
I was really fucking things up trying to squash commits. *uncomfortable feeling

**Pull request closed without merging**